pac4j-kerberos/src/main/java/org/pac4j/kerberos/credentials/authenticator/SunJaasKerberosTicketValidator.java

```diff
@@ -0,0 +1,247 @@
+package org.pac4j.kerberos.credentials.authenticator;
+
+import java.io.File;
+import java.security.Principal;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.auth.kerberos.KerberosPrincipal;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSCredential;
+import org.ietf.jgss.GSSException;
+import org.ietf.jgss.GSSManager;
+import org.ietf.jgss.GSSName;
+import org.pac4j.core.exception.BadCredentialsException;
+import org.pac4j.core.io.Resource;
+import org.pac4j.core.util.CommonHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of {@link KerberosTicketValidator} which uses the SUN JAAS
+ * login module, which is included in the SUN JRE, it will not work with an IBM JRE.
+ * The whole configuration is done in this class, no additional JAAS configuration
+ * is needed.
+ *
+ * @author Garry Boyce
+ * @since 1.9.1
+ * 
+ *  originally from spring-kerberos project
+ */
+public class SunJaasKerberosTicketValidator implements KerberosTicketValidator {
+
+    private String servicePrincipal;
+    private Resource keyTabLocation;
+    private Subject serviceSubject;
+    private boolean holdOnToGSSContext;
+    private boolean debug = false;
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    @Override
+    public KerberosTicketValidation validateTicket(byte[] token) {
+        try {
+            return Subject.doAs(this.serviceSubject, new KerberosValidateAction(token));
+        }
+        catch (PrivilegedActionException e) {
+            throw new BadCredentialsException("Kerberos validation not successful", e);
+        }
+    }
+
+
+    public void afterPropertiesSet() throws Exception {

```

Comment:
This won't work as is. The `afterPropertiesSet` is a Spring method for a bean implementing `InitializingBean`. In our case, we must use the `InitializableObject` or `InitializableWebObject` abstract classes and ensure the `internalInit`  method has been called from the authenticator.